apps/src/code-studio/progressRedux.js

```diff
@@ -84,18 +85,51 @@ const initialState = {
 /**
  * Thunks
  */
+
+// The user has navigated to a new level in the current lesson,
+// so we should update the browser and also set this as the new
+// current level.
 export function navigateToLevelId(levelId) {
   return (dispatch, getState) => {
     const state = getState().progress;
-    const newLevel = state.lessons[0].levels.find(level =>
-      level.ids.find(id => id === levelId)
-    );
+    const newLevel = state.lessons
+      .find(lesson => lesson.id === state.currentLessonId)
```

Comment:
minor: handle the case where `.find()` returns null?

Reply:
Agreed that this should be handled somehow, but also kind of torn, because if we handle it silently on the client only, we essentially mask an inability to report a milestone.  If we were to report it, we'd use one of the available reporting mechanisms; you probably have some ideas given your other current work?

Given that we deploy so often, I'd think that New Relic alerting us to the JS error would actually be a very effective way to catch an issue here, except that we have the issue with so many errors in there that we don't get very effective alerting from it...

Reply:
Yeah, if we want to flag this as invalid state from the server somehow, then logging would be best. If we want to take advantage of the new browser logging, maybe we can just create a TODO and link a task here?

Reply:
Opened a tracking issue [here](https://codedotorg.atlassian.net/browse/SL-823).